Tools/gdb/libpython.py

```diff
@@ -1797,14 +1797,14 @@ def is_gc_collect(self):
 
     def get_pyop(self):
         try:
-            frame = self._gdbframe.read_var('frame')
-            frame = PyFramePtr(frame)
-            if not frame.is_optimized_out():
-                return frame
+            # frame = self._gdbframe.read_var('frame')
+            # frame = PyFramePtr(frame)
+            # if not frame.is_optimized_out():
+            #     return frame
             cframe = self._gdbframe.read_var('cframe')
             if cframe is None:
                 return None
-            frame = PyFramePtr(cframe["current_frame"].dereference())
+            frame = PyFramePtr(cframe["current_frame"])
```

Comment:
This seems more correct. What does it fix?

Reply:
The pointer that we are getting was wrong. Seems that new versions of gdb still correct this fall and doesn't do anything if is a value and not a pointer, but old versions like the one  FreeBSD get garbage